crates/ruff_linter/src/rule_selector.rs

```diff
@@ -226,7 +225,8 @@ impl RuleSelector {
             // is turned on
             || ((rule.is_preview() || rule.is_nursery()) && preview_enabled && (self.is_exact() || !preview_require_explicit))
             // Deprecated rules are excluded in preview mode unless explicitly selected
-            || (rule.is_deprecated() && (!preview_enabled || self.is_exact()))
+            // Ignore deprecated rules for All option
+            || (rule.is_deprecated() && (!preview_enabled || self.is_exact()) && !matches!(self, RuleSelector::All { .. }))
```

Comment:
Can you combine this comment with the one above? e.g. 

> Deprecated rules are excluded from ALL in stable require exact selection in preview

Reply:
Did it!